examples/wakaama/dtlsconnection.c

```diff
@@ -0,0 +1,570 @@
+/*******************************************************************************
+ *
+ * Copyright (c) 2015 Intel Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * The Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ *    David Navarro, Intel Corporation - initial API and implementation
+ *    Christian Renz - Please refer to git log
+ *    Christian Manal - Ported to RIOT OS
+ *
+ *******************************************************************************/
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Handles DTLS/networking of the Wakaama example client
+ *
+ * @author      Christian Manal <manal@uni-bremen.de>
+ *
+ * @}
+ */
+
+#include <ctype.h>
+#include <stdlib.h>
+#include <string.h>
+#include "dtlsconnection.h"
+
+#define COAP_PORT "5683"
+#define COAPS_PORT "5684"
+#define URI_LENGTH 256
+
+static dtls_context_t *dtlsContext;
+
+static int gnrc_sending(char *addr_str, char *data, size_t data_len,
```

Comment:
This is working, but is legacy and is one of the main things I want to remove with PR #7615

Reply:
Will GNRC be deprecated or is it ok if it stays like this for now? I'm planning to port the Wakaama server as well in the future, so I could use gnrc_sock_udp for that and move the client over as further down the line.

Reply:
GNRC per se will not be deprecated, so far as I know. However, my implementation of that function is not very optimal (just look to that timer :( ).

The part of UDP_sock in the PR #7615 remove  that issue (I still need a delay for the client, but can be as low as 100 uS).   So, I strongly suggest to  upgrade this part while your PR is not merged.

Reply:
I see your point. I'll try to get to that next week.

Reply:
I've moved to socks now and the thing compiles and runs on native, but it doesn't seem to use the tap interface anymore, so I can't really test it. Is that expected behavior, or did I make a mistake somewhere?

Reply:
After banging my head against a wall for several days, it turns out that the socks API doesn't like sending packets when there's no netif ID set on the remote endpoint and the target IPv6 address is link-local. `sock_udp_send()` does not treat that as an error though, and in fact queues a packet that will (permanently ?) clog up the send queue.

I built in a workaround for that now.